examples/audio/eeg_signal.py

```diff
@@ -0,0 +1,527 @@
+"""
+Title: EEG Signal Classification for Stimuli identification
+Author: [Suvaditya Mukherjee](https://twitter.com/halcyonrayes)
+Date created: 2022/11/03
+Last modified: 2022/11/04
+Description: Training a Convolutional model to classify EEG signals produced by exposure to certain stimuli.
+"""
+"""
+## Introduction
+
+The following example explores how we can make a Convolution-based Neural Network to
+perform classification on EEG signals captured when subjects were exposed to different
+stimuli.
+We train a model from scratch since such signal-classification models are fairly scarce
+in pre-trained format.
+The data we use is sourced from the UC Berkeley-Biosense Lab where the data was collected
+from 15 subjects at the same time.
+Our process is as follows:
+
+- Load the [UC Berkeley-Biosense Synchronized Brainwave
+Dataset](https://www.kaggle.com/datasets/berkeley-biosense/synchronized-brainwave-dataset)
+- Visualize random samples from the data
+- Pre-process, collate and scale the data to finally make a `tf.data.Dataset`
+- Prepare class weights in order to tackle major imbalances
+- Create a Conv1D and Dense-based model to perform classification
+- Define callbacks and hyperparameters
+- Train the model
+- Plot metrics from History and perform evaluation
+
+This example needs the following external dependencies (gdown, Scikit-learn, Pandas,
+NumPy, Matplotlib). You can install it via the following commands.
+"""
+
+"""
+## Setup and Import headers
+"""
+
+"""shell
+pip install gdown
+pip install gdown
+pip install sklearn
+pip install pandas
+pip install numpy
+pip install matplotlib
+gdown 1V5B7Bt6aJm0UHbR7cRKBEK8jx7lYPVuX
```

Comment:
does gdown just install all of the data from google drive?  Maybe this deserves its own cell to explain what this command does to users - as it will take a lot of space on their PC.

Reply:
The `gdown` package takes up somewhere around 14.6kB and the actual `eeg-data.csv` is around 2mB in size. But I'll nevertheless add the cell with the information.

Reply:
I was mistaken, eeg-data.csv is actually around 105.7mB.